src/test/java/spleetwaise/transaction/logic/commands/AddCommandTest.java

```diff
@@ -89,68 +69,47 @@ public void execute_validTransaction_success() {
     }
 
-    @Test
-    public void execute_invalidPersonIndex_throwsCommandException() {
-        Index invalidIndex = Index.fromZeroBased(TypicalPersons.getTypicalAddressBook().getPersonList().size());
-        AddCommand cmd = new AddCommand(invalidIndex, testAmount,
-                testDescription, testDate, testCategories
-        );
-
-        Assert.assertThrows(CommandException.class, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX,
-                cmd::execute
-        );
-    }
-
     @Test
     public void equals_sameTransaction_returnsTrue() {
-        AddCommand cmd1 = new AddCommand(testIndex, testAmount, testDescription, testDate, testCategories);
-        AddCommand cmd2 = new AddCommand(testIndex, testAmount, testDescription, testDate, testCategories);
+        AddCommand cmd1 = new AddCommand(testTxn);
+        AddCommand cmd2 = new AddCommand(testTxn);
 
         assertEquals(cmd1, cmd1);
         assertEquals(cmd1, cmd2);
-        assertEquals(cmd1.toString(), cmd1.toString());
+        assertEquals(cmd1.toString(), cmd2.toString());
         assertEquals(cmd2.toString(), cmd2.toString());
     }
 
     @Test
     public void equals_diffTransaction_returnsFalse() {
-        AddCommand cmdBase = new AddCommand(testIndex, testAmount, testDescription, testDate, testCategories);
-
-        Index newIndex = Index.fromOneBased(2);
-        AddCommand cmdOther = new AddCommand(newIndex, testAmount, testDescription, testDate, testCategories);
-        assertNotEquals(cmdBase, cmdOther);
-
-        Amount newAmount = new Amount("12.3");
-        cmdOther = new AddCommand(testIndex, newAmount, testDescription, testDate, testCategories);
-        assertNotEquals(cmdBase, cmdOther);
-
-        Description newDesc = new Description("different description");
-        cmdOther = new AddCommand(testIndex, testAmount, newDesc, testDate, testCategories);
-        assertNotEquals(cmdBase, cmdOther);
-
-        Date newDate = new Date("01022024");
-        cmdOther = new AddCommand(testIndex, testAmount, testDescription, newDate, testCategories);
-        assertNotEquals(cmdBase, cmdOther);
-
-        cmdOther = new AddCommand(testIndex, testAmount, testDescription, testDate,
-                new HashSet<>(List.of(new Category("EXTRA")))
-        );
-        assertNotEquals(cmdBase, cmdOther);
+        AddCommand cmd1 = new AddCommand(testTxn);
+        Transaction testTxn2 = new Transaction(
+                TypicalPersons.BOB, testAmount, testDescription, testDate, testCategories);
+        AddCommand cmd2 = new AddCommand(testTxn2);
+        Transaction testTxn3 = new Transaction(
+                TypicalPersons.BOB, testAmount, testDescription, testDate,
+                new HashSet<>(List.of(new Category("EXTRA"))));
+        AddCommand cmd3 = new AddCommand(testTxn3);
+
+        assertNotEquals(cmd1, cmd2);
+        assertNotEquals(cmd1, cmd3);
+        assertNotEquals(cmd2, cmd3);
```

Comment:
These two assertions here are unnecessary as `assertNotEquals(cmd1, cmd2);` has already checked that different commands should not be equal.